core/contract.go

```diff
@@ -0,0 +1,49 @@
+package core
+
+import (
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+// Class unambiguously defines a [Contract]'s semantics.
+type Class struct {
+	// The version of the class, currently always 0.
+	APIVersion uint
+	// External functions defined in the class.
+	Externals []EntryPoint
+	// Functions that receive L1 messages. See
+	// https://www.cairo-lang.org/docs/hello_starknet/l1l2.html#receiving-a-message-from-l1
+	L1Handlers []EntryPoint
+	// Constructors for the class. Currently, only one is allowed.
+	Constructors []EntryPoint
+	// An ascii-encoded array of builtin names imported by the class.
+	Builtins []string
+	// The starknet_keccak hash of the JSON compiler output.
+	ProgramHash []byte
```

Comment:
Do you think it would be good to use `*felt.Felt`?

Reply:
`sn_keccak` is a modified Keccak256 which outputs a value in the same field Pedersen operates on. So It probably should be a `*felt.Felt`

https://docs.starknet.io/documentation/develop/Hashing/hash-functions/#starknet_keccak

Reply:
Nice catch, thanks!